src/js/EditorBlock.js

```diff
@@ -84,7 +99,47 @@ export default class EditorBlock extends Component {
             )
         }
 
-        const keyInput = (
+        const userAPIKeyMessage = this.state.isSavedKey ? (
+            <div>
+                <p style={{textAlign: 'center'}}>
+                    {__( 'Only administrators can change the Google Maps API key.')  }<br />
+                    {__( 'Note: changing the API key effects all Google Map Embed blocks.')  }
+                </p>
+                <TextControl
+                    key="api-input"
+                    value={ this.state.apiKey }
+                    onChange={ value => this.setState({ apiKey: value }) }
+                    style={{textAlign: 'center', border: 'solid 1px rgba(100,100,100,0.25)'}}
+                    readOnly={true}
+                    placeholder={ __('API Key') }
+                />
+                <p style={{textAlign: 'center', paddingBottom: '1em'}}>
+                    {__('Need an API key? Get one')}&nbsp;
+                    <a href="https://console.developers.google.com/flows/enableapi?apiid=maps_backend,static_maps_backend,maps_embed_backend&keyType=CLIENT_SIDE&reusekey=true">
+                        {__('here.')}
+                    </a><br /><br />
+                    <Button 
+                        isPrimary 
+                        onClick={ this.saveApiKey }
+                        isBusy={ this.state.isSaving }
+                        disabled={ true }
+                    >
+                        {__('Save API key')}
+                    </Button>
+                </p>
+            </div>
+        ) : (
+            <div>
+                <p style={{textAlign: 'center'}}>
+                    {__( 'A Google Maps API key is required to use the map block, please ask an administrator to enter one. An API key can be obtained')  }&nbsp;
```

Comment:
Links that just say `here` aren't great for accessibility, I'd move the preceding text "An API key can be obtained" into the link as well.